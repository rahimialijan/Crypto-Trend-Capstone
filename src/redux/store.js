import { configureStore } from "@reduxjs/toolkit";
import cryptoReducer from "./crypto/cryptoSlice";
const REDUX_DEVTOOLS = true;

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
  devTools: REDUX_DEVTOOLS || false,
});

export default store;
