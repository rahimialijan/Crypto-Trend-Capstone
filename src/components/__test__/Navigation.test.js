import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import createMockStore from 'redux-mock-store';
import Nav from '../Navigation/Navigation';
import store from '../../redux/store';

describe('Navigation Bar test suit', () => {
  const initialState = {
    crypto: {
      cryptos: [
        {
          uuid: 'Qwsogvtv82FCd',
          symbol: 'BTC',
          name: 'Bitcoin',
          color: '#f7931A',
          iconUrl: 'https://cdn.coinranking.com/bOabBYkcX/bitcoin_btc.svg',
          marketCap: '586699044075',
          price: '30525.981366112937',
          listedAt: 1330214400,
          tier: 1,
          change: '0.50',
          rank: 1,
          lowVolume: false,
          coinrankingUrl: 'https://coinranking.com/coin/Qwsogvtv82FCd+bitcoin-btc',
          '24hVolume': '12581186855',
          btcPrice: '1',
        },
      ],
    },
    stats: {
      total: 26788,
      totalCoins: 26788,
      totalMarkets: 36444,
      totalExchanges: 161,
      totalMarketCap: '1172476637995',
      total24hVolume: '35484231592',
    },
    title: 'Crypto Trend',
    isLoading: true,
    error: '',
    message: '',
    active: 'Qwsogvtv82FCd',
  };
  const mockStore = createMockStore();
  const storeMocked = mockStore(initialState);
  test('Should render a navigation bar', () => {
    const container = render(
      <Provider store={storeMocked}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Should render Title in navigation bar', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.getByText(/Crypto Trend/i)).toBeInTheDocument();
  });
});
