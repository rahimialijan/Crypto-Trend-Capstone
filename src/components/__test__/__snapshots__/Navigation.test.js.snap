// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar test suit Should render a navigation bar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <span
          class="logo"
        >
          <a
            href="/"
          >
            <img
              alt="logo"
              class="back-button"
              src="back.svg"
            />
          </a>
        </span>
        <span
          class="nav-title"
        />
        <span
          class="nav-icon"
        >
          <img
            alt=""
            src="microphone.svg"
          />
          <img
            alt=""
            src="setting.svg"
          />
        </span>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <span
        class="logo"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="back-button"
            src="back.svg"
          />
        </a>
      </span>
      <span
        class="nav-title"
      />
      <span
        class="nav-icon"
      >
        <img
          alt=""
          src="microphone.svg"
        />
        <img
          alt=""
          src="setting.svg"
        />
      </span>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
